### PR TITLE
Simulated time trimming & profiler python binding adjustments

### PR DESCRIPTION
**Pull request by yodada**
In this PR:
 - provides a way to "trim" execution time profiling (@vb000 I think this is where your bsg_time goes)
 - re-arrange profiler python bindings. Now we operates from `torch.hammerblade.profiler`

### CODE REVIEW

File: c10/hammerblade/HammerBladeFunctions.cpp [hunk 1]
```diff
@@ -75,10 +75,19 @@ void offload_kernel(const char* kernel, std::vector<eva_t> args) {
...
 
+  // create a trim log so we can write logs directly
+  c10::probe::HBProfilerTrimLog* trim_log = new c10::probe::HBProfilerTrimLog();
+
   C10_HB_CHECK(hb_mc_kernel_enqueue(&_hb_device, _hb_grid_dim, _hb_tg_dim, kernel,
                                     args.size(), cuda_argv));
   C10_HB_CHECK(hb_mc_device_tile_groups_execute(&_hb_device));
 
+  // write the SIMULATED time to ExecutionTime log
+  std::chrono::microseconds simulated(3154);
```
Comment (yodada) [inline c10/hammerblade/HammerBladeFunctions.cpp:111]: A magic number here ... just to say this mechanism works

**Comment by vb000** [comment]
I managed to try it out but seeing this error... am I doing something wrong?
```python
import torch
import torch.hammerblade.profiler as hbprof

x = torch.rand(2,3).hammerblade()
y = torch.rand(2,3).hammerblade()

hbprof.enable()

print(x + y)

hbprof.exec_time.fancy_print()
```
```
Traceback (most recent call last):
  File "tmp.py", line 11, in <module>
    hbprof.exec_time.fancy_print()
  File "/mnt/users/ssd1/no_backup/bandhav/hb-pytorch/torch/hammerblade/profiler/exec_time.py", line 124, in fancy_print
    root = exec_time_tree(fancy_func=True, trimming=trimming)
  File "/mnt/users/ssd1/no_backup/bandhav/hb-pytorch/torch/hammerblade/profiler/exec_time.py", line 111, in exec_time_tree
    root = exec_time_Node(roi[0], roi[1])
IndexError: tuple index out of range
```

**Comment by yodada** [comment]
Ok! added a new top level wrapper `torch.hammerblade.profiler.stats()`
You may refer to the README for how to use these tools. (should be clear ... I hope)

**Comment by yodada** [comment]
@vb000 take another look when you have time ;)

**Review by vb000 — APPROVED** [review]
Looks good... thanks for the explanation!

Sorry for these nitpicks, I couldn't stop myself from thinking about these: if `fancy_print` is the only print in the module why not just name is `print`? Easier to remember ;). Similarly `add_beacon` could just be `add`?

**Comment by vb000** [comment]
On a serious note though, I think `chart.fancy_print()` could be `chart.json()` hinting user that it's intended to be written to a json file.

**Comment by yodada** [comment]
As you wish!
`chart.add_beacon` -> `chart.add`
`chart.clear_beacon` -> `chart.clear`
`chart.fancy_print` -> `chart.json`
`route.add_waypoint` -> `route.add`
`route.fancy_print` -> `route.json`

**Pull request merged**